### PR TITLE
fix(rust, python) validate utc, fmt, and tz-aware in strptime

### LINKED ISSUE #6549: ComputeError not raised if utc=True and fmt=None
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Issue description

`utc` is only meant to be used if `tz_aware=True` (or if tz-awareness is inferred), but passing it when `fmt=None` has no effect

### Reproducible example

```
In [1]: pl.Series(["2020-01-01 03:00:00"]).str.strptime(pl.Datetime, utc=True)
Out[1]:
shape: (1,)
Series: '' [datetime[μs]]
[
        2020-01-01 03:00:00
]
```

### Expected behavior

```
ComputeError: Cannot use 'utc=True' with tz-naive data. Parse the data as naive, and then use `.dt.with_time_zone('UTC')
```

### Installed versions

<details>
---Version info---
Polars: 0.16.1
Index type: UInt32
Platform: Linux-5.10.102.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python: 3.8.16 (default, Dec  7 2022, 01:12:06)
[GCC 11.3.0]
---Optional dependencies---
pyarrow: 10.0.1
pandas: 1.4.4
numpy: 1.23.5
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: <not installed>
deltalake: <not installed>
matplotlib: 3.6.2
</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #6549 

it should be possible to auto-detect tz-aware if `fmt` is None, but I'll try that separately if that's OK - for now I'd suggest just throwing an informative error

### CODE REVIEW

File: polars/polars-lazy/polars-plan/src/dsl/function_expr/strings.rs [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 
+#[cfg(feature = "regex")]
+use once_cell::sync::Lazy;
 use polars_arrow::utils::CustomIterTools;
 #[cfg(feature = "regex")]
 use regex::{escape, Regex};
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
+#[cfg(feature = "regex")]
```
Comment (ritchie46) [inline polars/polars-lazy/polars-plan/src/dsl/function_expr/strings.rs:11]: I think this should be under feature `timezones` and then the `timezones` feature can activate `regex`.

File: polars/polars-lazy/polars-plan/src/dsl/function_expr/strings.rs [hunk 2]
```diff
@@ -1,11 +1,16 @@
 use std::borrow::Cow;
 
+#[cfg(feature = "regex")]
```
Comment (ritchie46) [inline polars/polars-lazy/polars-plan/src/dsl/function_expr/strings.rs:3]: I think this should be under feature `timezones` and then the `timezones` feature can activate `regex`.

File: polars/polars-lazy/polars-plan/src/dsl/function_expr/strings.rs [hunk 3]
```diff
@@ -308,6 +313,23 @@ pub(super) fn count_match(s: &Series, pat: &str) -> PolarsResult<Series> {
...
+            return Err(PolarsError::ComputeError(
+                "Passing 'tz_aware=True' without 'fmt' is not yet supported. Please specify 'fmt'."
+                    .into(),
+            ));
+        }
+        #[cfg(feature = "regex")]
+        (false, Some(fmt)) => TZ_AWARE_RE.is_match(fmt),
+        (false, _) => false,
+    };
+    if !tz_aware && options.utc {
```
Comment (ritchie46) [inline polars/polars-lazy/polars-plan/src/dsl/function_expr/strings.rs:328]: This check should be begind `timezones` feature flag.

**Review by ritchie46 — COMMENTED** [review]
Thanks @MarcoGorelli. Small comment about the feature flags, other than that good to go.

**Comment by ritchie46** [comment]
@MarcoGorelli you want to change this in `polars-plan`:

from 
```
timezones = ["polars-time/timezones", "polars-core/timezones"]
```

to 
```
timezones = ["polars-time/timezones", "polars-core/timezones", "regex"]
```

**Comment by MarcoGorelli** [comment]
> 

got it - thanks 🙌 !

**Pull request merged**